lib/exitCodes.ts

```diff
@@ -1,28 +1,79 @@
 import {Logger} from './logger2';
 
 const CONFIG_ERROR_CODE = 105;
+const BROWSER_ERROR_CODE = 135;
+
+export class ProtractorError extends Error {
+  static ERR_MSGS: string[];
+  static DEFAULT_MSG = 'protractor encountered an error';
 
-export class ProtractorError {
   error: Error;
   description: string;
   code: number;
   stack: string;
   constructor(logger: Logger, description: string, code: number) {
+    super();
     this.error = new Error();
-    this.description = description;
     this.code = code;
+    this.description = description;
+    this.stack = this.error.stack;
+    this.logError(logger);
+  }
+
+  logError(logger: Logger) {
     logger.error('error code: ' + this.code);
     logger.error('description: ' + this.description);
-    this.stack = this.error.stack;
+    logger.error(this.stack);
   }
 }
-
 /**
  * Configuration file error
  */
 export class ConfigError extends ProtractorError {
+  static DEFAULT_MSG = 'configuration error';
   static CODE = CONFIG_ERROR_CODE;
-  constructor(logger: Logger, description: string) {
-    super(logger, description, ConfigError.CODE);
+  constructor(logger: Logger, opt_msg?: string) {
+    super(logger, opt_msg || ConfigError.DEFAULT_MSG, ConfigError.CODE);
+    process.exit(ConfigError.CODE);
+  }
+}
+
+/**
+ * Browser errors including getting a driver session, direct connect, etc.
+ */
+export class BrowserError extends ProtractorError {
+  static DEFAULT_MSG = 'browser error';
```

Comment:
Ditto for not having a default error, make the error-thrower say something useful.